zvariant/src/ser.rs

```diff
@@ -208,11 +208,10 @@ where
 /// Serialize `T` that has the given signature, to a new byte vector.
 ///
 /// Use this function instead of [`to_bytes`] if the value being serialized does not implement
-/// [`DynamicType`]. See [`from_slice_for_signature`] documentation for an example of how to use
+/// [`DynamicType`]. See [`deserialize_for_signature`] documentation for an example of how to use
```

Comment:
We could just make this [`Data::deserialize_for_signature`] since `Data` is already in the scope. A lot of what you're updating is from the time before you could directly link like this and all the new doc comments make heavy use of this directly linking so might be good to make things consistent and simpler while you're at it?